langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCacheType.java

```diff
@@ -0,0 +1,56 @@
+package dev.langchain4j.model.anthropic.internal.api;
+
+import dev.langchain4j.model.anthropic.internal.api.AnthropicMessageContent.CacheControl;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.function.Supplier;
+
+@Getter
+public class AnthropicCacheType {
```

Comment:
Perharps we can simplify this? It will ofc depend on the final implementaiton.

As a user, I would expect to be able to:
- Cache system message (I want to put a lot of instructions/info in the system message). In this case I would like to toggle `.cacheSystemMessage(true)` when building `AnthropicChatModel` and expect that `"cache_control": {"type": "ephemeral"}` is inserted after the system message. In this case I will use the same system message and not change it between calls to the LLM.
- Cache tools (I have lots of tools and they take a lot of tokens). In this case I would like to toggle `.cacheTools(true)` when building `AnthropicChatModel` and expect that `"cache_control": {"type": "ephemeral"}` is inserted after the last tool. In this case I will use the same list of tools and not change them between calls to the LLM.
- Cache user messages. Here I am not fully sure how exactly it should work. Should `"cache_control": {"type": "ephemeral"}` be inserted after each user message? Or only after the last one? What about limit of 4 cache blocks?

Please notice that user should be able to enable caching of system message, tools and user messages independently.
Existing options (ALL, NONE, SYSTEM_MESSAGE, USER_MESSAGE) do not seem to support this.